code/modules/power/cable.dm

```diff
@@ -61,23 +52,25 @@ GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/gri
 	var/under_thing = NONE
 	if(clear_before_updating)
 		linked_dirs = 0
+
 	var/obj/machinery/power/search_parent
-	for(var/obj/machinery/power/P in loc)
-		if(istype(P, /obj/machinery/power/terminal))
+	for(var/obj/machinery/power/power_machine in associated_loc)
+		if(istype(power_machine, /obj/machinery/power/terminal))
 			under_thing = UNDER_TERMINAL
-			search_parent = P
+			search_parent = power_machine
 			break
-		if(istype(P, /obj/machinery/power/smes))
+		if(istype(power_machine, /obj/machinery/power/smes))
 			under_thing = UNDER_SMES
-			search_parent = P
+			search_parent = power_machine
 			break
+
 	for(var/check_dir in GLOB.cardinals)
-		var/TB = get_step(src, check_dir)
+		var/turf/step_turf = get_step(associated_loc, check_dir)
 		//don't link from smes to its terminal
 		if(under_thing)
 			switch(under_thing)
 				if(UNDER_SMES)
-					var/obj/machinery/power/terminal/term = locate(/obj/machinery/power/terminal) in TB
+					var/obj/machinery/power/terminal/term = locate(/obj/machinery/power/terminal) in step_turf
```

Comment:
not gonna undertile terminals I take it?

Reply:
nope